cluster-autoscaler/simulator/cluster.go

```diff
@@ -117,79 +117,56 @@ func FindNodesToRemove(
 	listers kube_util.ListerRegistry,
 	clusterSnapshot ClusterSnapshot,
 	predicateChecker PredicateChecker,
-	maxCount int,
-	fastCheck bool,
```

Comment:
How are Fast and Detailed backed by the same checks? Fast calls `GetPodsForDeletionOnNodeDrain` with `checkReferences=false`, while Detailed calls it with `checkReferences=true`. `checkReferences` determines whether `GetPodsForDeletionOnNodeDrain` does GETs on pod controllers. I'm not sure what the actual difference in performance this implies, but it looks like a material change - or are you saying that the GETs don't matter?

Reply:
I don't think this is actually generating any GET requests, CA is only fetching cached objects from listers wheck `checkReferences=true`. To be honest I'm not sure if that flag makes sense anymore.

Reply:
The listers are backed up by the cache so there should not be a noticeable number of new GET calls.
I did run preliminary 5k test to verify and I did not see any regressions.

Reply:
I've re-run the tests with current version and still did not find any increase in number of GAT calls.

Reply:
Ok, thanks for verifying that! We should get rid of the parameter altogether then, but that's probably out-of-scope of this PR.